Bio/Phenotype/PhenMicro.py

```diff
@@ -0,0 +1,1101 @@
+
+        Note that long well lists are shown truncated.
+        """
+        lines = []
+        if self.id:
+            lines.append("Plate ID: %s" % self.id)
+        lines.append("Well: %i" % len(self))
+        # Here we assume that all well ID start with a char
+        lines.append("Rows: %d" % 
+                    len(set([x.id[0] for x in self])))
+        # Here we assume that well number is a two-digit number
+        lines.append("Columns: %d" % 
+                    len(set([x.id[1:3] for x in self])))
+        lines.append(repr(self))
+        return "\n".join(lines)
+
+class WellRecord(object):
+    """WellRecord object stores all the time course signals of a Phenotype
+    Microarray well.
+    
+    The single time points and signals can be
+    accessed iterating on the WellRecord or using lists indeces or slices:
+    
+    >>> from Bio import Phenotype
+    >>> plate = Phenotype.read("plate.csv", "pm-csv")
+    >>> well = plate['A05']
+    >>> for time, signal in well:
+    ...    print("Time: %f, Signal: %f" % (time, signal))
+    Time: 0.0, Signal: 10.0
+    Time: 0.25, Signal: 14.0
+    Time: 0.5, Signal: 19.0
+    Time: 24.25, Signal: 32.0
+    ...
+    >>> well[1]
+    23.0
+    >>> well[1:5]
+    [23.0, 23.0, 26.0, 26.0]
+    >>> well[1:5:0.5]
+    [23.0, 23.0, 23.0, 26.0, 26.0, 26.0, 26.0, 26.0]
+    
+    If a time point was not present in the input file but it's between the
+    minimum and maximum time point, the interpolated signal is returned, 
+    otherwise a nan value:
+    
+    >>> well[1.3]
+    23.0
+    >>> well[1250]
+    nan
+    
+    Two WellRecord objects can be compared: if their input time/signal pairs
+    are exactly the same, the two records are considered equal:
+    
+    >>> well2 = plate['H12']
+    >>> well == well2
+    False
+    
+    Two WellRecord objects can be summed up or subtracted from each other: a new
+    WellRecord object is returned, having the left operand id.
+    
+    >>> well2 = well + well1
+    >>> print(well2.id)
+    A05
+    
+    If SciPy is installed, a sigmoid function can be fitted to the PM curve,
+    in order to extract some parameters; three sigmoid functions are available:
+    * gompertz
+    * logistic
+    * richards
+    The functions are described in Zwietering et al., 1990 (PMID: 16348228)
+    
+    >>> well.fit()
+    >>> print(well.slope, well.model)
+    (61.853516785566917, 'logistic')
+    
+    If not sigmoid function is specified, the first one that is succesfully
+    fitted is used. The user can also specify a specific function.
+    
+    >>> well.fit('gompertz')
+    >>> print(well.slope, well.model)
+    (127.94630059171354, 'gompertz')
+    
+    If no function can be fitted, the parameters are left as None, except for
+    the max, min, average_height and area. 
+    """
+    
+    def __init__(self, wellid, plate=PlateRecord(None), signals={}):
```

Comment:
Never use a mutable object as a default argument (see eg https://pythonconquerstheuniverse.wordpress.com/category/python-gotchas/ for why), use ``None`` instead and then perhaps:

```
def __init__(self, wellid, plate=None, signals={}):
    # your code here
    if plate is None:
        self.plate = PlateRecord(None)
    else:
        self.plate = None
    if signals is None:
        self._signals = {}
    else:
        self._signals = signals
```